### PR TITLE
[Logs Agent Component] Migrate `pkg/logs/config` to `comp/logs/agent/config`

### PR DESCRIPTION
**Pull request by gh123man**
### What does this PR do?

This PR is part 3 in a series of PRs to migrate the logs agent to a component. 

This PR migrates `pkg/logs/config` to `comp/logs/agent/config`. In order to make this move compatible with the component framework, the various functions in this package needed to be updated to take an explicit config instance instead of relying on the global config. As a result, the PR refactors the cases where the logs config is used to pass in the config. 

The next PR will contain the final top-level logs agent migration that leverages these changes. 

### Logs agent component roadmap

In order to make review easier, this migration is broken down into 4 PRs:

✅ [define the component interface](https://github.com/DataDog/datadog-agent/pull/18584)
✅ [Move launchers, tailers, and metrics out of internal](https://github.com/DataDog/datadog-agent/pull/18646)
👉 [Migrate `pkg/logs/config` to `comp/logs/agent/config`](https://github.com/DataDog/datadog-agent/pull/18691) 
🕒 Migrate top level logs implementation to `comp`

### Describe how to test/QA your changes

There should be no functional change, but it would be good to spot check/validate that basic logs agent scenarios work, as well as for other teams this change impacts to spot check that their code paths using the logs config still behave correctly. 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]
Approving for changes in `pkg/otlp`

**Review by remeh — APPROVED** [review]
:shipit:

File: pkg/config/config.go [hunk 1]
```diff
@@ -1549,6 +1549,7 @@ func LoadDatadogCustom(config Config, origin string, loadSecret bool, additional
 	}
 
 	SanitizeAPIKeyConfig(config, "api_key")
+	SanitizeAPIKeyConfig(config, "logs_config.api_key")
```
[the remark this answers is not in the document]
Comment (gh123man) [reply pkg/config/config.go:1552]: I moved it [out from `BuildEndpoints`](https://github.com/DataDog/datadog-agent/pull/18691/files#diff-57088e26489f039750c3101e2dc58515805e6270672e208f2328171de319d0d6L98) for 2 reasons: 

- We can't call `SanitizeAPIKeyConfig` on a `ConfigReader`, and we need to use the `ConfigReader` interface in `comp` to not tightly couple to the global config. 
- I also think it makes more sense to do `SanitizeAPIKeyConfig` all in one place for global config keys.

[the remark this answers is not in the document]
Comment (remeh) [reply pkg/config/config.go:1552]: * Missed [this](https://github.com/DataDog/datadog-agent/pull/18691/files#diff-57088e26489f039750c3101e2dc58515805e6270672e208f2328171de319d0d6L98) indeed 👍 
* 🍰

**Review by paulcacheux — APPROVED** [review]

File: comp/logs/agent/config/config.go [hunk 2]
```diff
@@ -94,50 +94,47 @@ func HasMultiLineRule(rules []*ProcessingRule) bool {
...
 		"To benefit from increased reliability and better network performances, "+
 		"we strongly encourage switching over to compressed HTTPS which is now the default protocol.",
 		logsConfig.getConfigKey("force_use_tcp"), logsConfig.getConfigKey("socks5_proxy_address"))
-	return buildTCPEndpoints(logsConfig)
+	return buildTCPEndpoints(coreConfig, logsConfig)
 }
 
 // BuildServerlessEndpoints returns the endpoints to send logs for the Serverless agent.
-func BuildServerlessEndpoints(intakeTrackType IntakeTrackType, intakeProtocol IntakeProtocol) (*Endpoints, error) {
-	coreConfig.SanitizeAPIKeyConfig(coreConfig.Datadog, "logs_config.api_key")
```
Comment (IvanTopolcic) [inline comp/logs/agent/config/config.go:126]: Was this expected to be removed?

Comment (gh123man) [reply comp/logs/agent/config/config.go:126]: It was moved into `config.go`  -[ see the above comment for the reasoning ](https://github.com/DataDog/datadog-agent/pull/18691#discussion_r1288416792)

Let me know if you have any concerns.

**Review by IvanTopolcic — APPROVED** [review]

**Review by AliDatadog — APPROVED** [review]
👍 for files owned by us.

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**